core/src/main/scala/com/databricks/labs/remorph/parsers/tsql/TSqlRelationBuilder.scala

```diff
@@ -141,25 +141,104 @@ class TSqlRelationBuilder extends TSqlParserBaseVisitor[ir.Relation] {
       .map(alias => ir.TableAlias(ctx.tableName().accept(this), alias.id.getText))
       .getOrElse(ctx.tableName().accept(this))
 
-  private[tsql] def translateJoinType(ctx: JoinOnContext): ir.JoinType = ctx.joinType() match {
-    case jt if jt == null || jt.outerJoin() == null || jt.INNER() != null => ir.InnerJoin
-    case jt if jt.outerJoin().LEFT() != null => ir.LeftOuterJoin
-    case jt if jt.outerJoin().RIGHT() != null => ir.RightOuterJoin
-    case jt if jt.outerJoin().FULL() != null => ir.FullOuterJoin
-    case _ => ir.UnspecifiedJoin
+  private def buildJoinPart(left: ir.Relation, ctx: JoinPartContext): ir.Relation = {
+    ctx match {
+      case c if c.joinOn() != null => buildJoinOn(left, c.joinOn())
+      case c if c.crossJoin() != null => buildCrossJoin(left, c.crossJoin())
+      case c if c.apply() != null => buildApply(left, c.apply())
+      case c if c.pivot() != null => buildPivot(left, c.pivot())
+      case _ => buildUnpivot(left, ctx.unpivot()) // Only case left
```

Comment:
```suggestion
      case c if c.unpivot() != null => buildUnpivot(left, ctx.unpivot())
      case _ => ir.UnspecifiedJoin
```

Reply:
At the point of the `case _`, the only possibility is `unpivot()` and the `case _` cannot be reached. Is there a general scala convention to avoid using `case _` as I have done here?